public/app/core/services/NewFrontendAssetsChecker.ts

```diff
@@ -1,41 +1,48 @@
-import { isEqual } from 'lodash';
-
 import { getBackendSrv, getGrafanaLiveSrv } from '@grafana/runtime';
 
-import { config } from '../config';
-
 export class NewFrontendAssetsChecker {
-  private _hasUpdates = false;
+  private hasUpdates = false;
+  private previous = '?';
```

Comment:
this would mean every grafana instance will have to refresh at least once to fill out the `previous` state? So basically as soon as you open grafana it will always refresh for everyone?

Reply:
the actual logic makes sure previous has been called before, many ways to slice it, but ths avoids the case you are talking about
```
if (this.checked && this.previous !== result) {
```
torkels original version (the PR this is targeting) saves the original values directly in the bootdata -- but I'm not sure we need/want that.  This has the downside that the first navigation will *never* find a change, only the second